nym-api/src/coconut/keys/mod.rs

```diff
@@ -64,11 +64,11 @@ impl KeyPair {
         }
     }
 
-    pub async fn verification_key(&self) -> Option<RwLockReadGuard<'_, VerificationKey>> {
-        RwLockReadGuard::try_map(self.get().await?, |maybe_keypair| {
-            maybe_keypair.as_ref().map(|k| k.keys.verification_key())
-        })
-        .ok()
+    pub async fn verification_key(&self) -> Option<VerificationKeyAuth> {
```

Comment:
The reason I was returning `RwLockReadGuard` was to avoid key cloning for simple reads. And the key is quite complex -> it's bunch of curve points.

I think the issue here comes from the fact that `keys.verification_key()` in the crypto code does key cloning. Not sure if I left this comment in the previous PR